docs/compiler.rst

```diff
@@ -0,0 +1,273 @@
+Guide to the Scenic Parser & Compiler
+=====================================
+
+Architecture & Terminology
+--------------------------
+
+.. figure:: /images/parser_architecture.png
+  :alt: Scenic Parser & Compiler Architecture
+  :figclass: align-center
+
+Scenic AST
+~~~~~~~~~~
+
+
+Scenic AST is an abstract syntax tree for representing Scenic programs.
+It is a superset of Python AST and includes nodes for Scenic-specific
+language constructs.
+
+``syntax/ast.py`` defines all Scenic-specific AST nodes. Note that all
+Scenic AST nodes must extend the ``AST`` class defined on the same file.
```

Comment:
... defined [in] the same file.

Reply:
fixed 4cf64494a4ef4340c82445ac97db27fe9d1dd05b